2.0/flask_app/services/rag_agent.py

```
@@ -0,0 +1,392 @@
+        q = decision.rewritten_query or ""
+        optimized = self._optimize_query_for_embedding(q)
+        vec = self._embed(optimized)
+        where_sql, params = self._build_where(decision.filters)
+        sql = (
+            "SELECT fc.id as chunk_id, fc.filing_id, f.company_name, f.cik, f.filing_type, f.filing_date, "
+            "       f.sic, f.sic_desc, fc.chunk_index, fc.content, "
+            "       (fc.embedding <=> $%d) AS distance "
+            "FROM filing_chunks fc JOIN filings f ON f.id = fc.filing_id %s "
+            "ORDER BY fc.embedding <=> $%d ASC "
+            "LIMIT %d;"
+            % (len(params) + 1, where_sql, len(params) + 1, top_k)
```

Ruff (0.13.3)

Possible SQL injection vector through string-based query construction

(S608)